feat: 탐험 장소 수정 QueryDSL 사용

dain391 opened the pull request:
## 📌 개요
> 장소 수정 시, 요청으로 전달된 필드만 선택적으로 업데이트하도록 QueryDSL 적용
## ✅ 작업 사항
- PlaceRepositoryCustom, PlaceRepositoryImpl 추가
- updatePlace(placeId, userId, request) 구현
- PlaceRepository에 Custom 인터페이스 상속
- PlaceService.updatePlace()에서 Custom 메서드 호출하도록 수정
- 수정할 값이 없을 때 NO_FIELDS_TO_UPDATE 예외 반환

## 🔍 리뷰 요청 포인트

## 💬 기타 사항
- 관련 이슈: #100

dain391 committed:
feat: 탐험 장소 수정 QueryDSL 사용

- PlaceRepositoryCustom/PlaceRepositoryImpl 생성
- PlaceRepository에 Custom 인터페이스 상속
- ErrorCode에 NO_FIELDS_TO_UPDATE 추가

#100

dain391 committed:
feat: generated 중복 수정

#100

dain391 committed:
feat: build.gradle 수정

#100

dain391 committed:
Merge branch 'develop' into feature/refactorPlaceQueryDSL

GyeongSe99 commented on src/main/java/com/explorer/gabom/domain/place/repository/PlaceRepositoryImpl.java:
```diff
@@ -0,0 +1,35 @@
+
+	@Override
+	public Place updatePlace(Long placeId, Long userId, PlaceUpdateRequest request) {
+		QPlace place = QPlace.place;
+
+		long updated = queryFactory.update(place)
+								   .set(place.title, request.getTitle())
+								   .set(place.address, request.getAddress())
+								   .set(place.lat, request.getLat())
+								   .set(place.lng, request.getLng())
+								   .set(place.proofMethod, request.getProofMethod())
+								   .set(place.content, request.getContent())
+								   .where(place.id.eq(placeId).and(place.user.id.eq(userId)))
+								   .execute();
+
+		if (updated == 0) return null;
+
```
이렇게 되면 null인 필드는 null을 저장해버려서 null체크를 따로 해주셔야합니다!

GyeongSe99 replied:
아니면 PATH 말고 PUT으로 구현하는 방법도 있습니다

dain391 replied:
If문 처리 했습니다!
테스트 완료~~~~~~~~~~~~~~~~~~

dain391 committed:
feat: PlaceRepositoryImpl 동적 업데이트 처리

#100

dain391 committed:
fix: PlaceRepositoryImpl 순환 참조 제거

- PlaceRepositoryImpl에서 PlaceRepository 필드 주입 제거
- QueryDSL 전용 로직만 남기도록 구조 개선

#100
